Tests/PubNubTests/Integration/PresenceEndpointIntegrationTests.swift

```diff
@@ -0,0 +1,219 @@
+
+  func testHereNow_SingleChannel_Stateless() {
+    let hereNowExpect = expectation(description: "Here Now Response")
+
+    let configuration = PubNubConfiguration(from: testsBundle)
+    let client = PubNub(configuration: configuration)
+
+    let performHereNow = {
+      // Here Now Test
+      client.hereNow(on: [self.channel]) { result in
+        switch result {
+        case let .success(response):
+          let channelPresence = response.channels[self.channel]
+          XCTAssertNotNil(channelPresence)
+        case let .failure(error):
+          XCTFail("Failed due to error: \(error)")
+        }
+        hereNowExpect.fulfill()
+      }
+    }
+
+    let listener = SubscriptionListener()
+    listener.didReceivePresence = { event in
+      if event.channel == self.channel, event.join.contains(configuration.uuid) {
+        performHereNow()
+      }
+    }
+    let token = client.add(listener)
+
+    client.subscribe(to: [channel], withPresence: true)
+
+    defer { token.cancel() }
+    wait(for: [hereNowExpect], timeout: 10.0)
+  }
+
+  func testHereNow_SingleChannel_State() {
+    let hereNowExpect = expectation(description: "Here Now Response")
+
+    let configuration = PubNubConfiguration(from: testsBundle)
+    let client = PubNub(configuration: configuration)
+
+    let performHereNow = {
+      // Here Now Test
+      client.hereNow(on: [self.channel], also: true) { result in
+        switch result {
+        case let .success(response):
+          let channel = response.channels[self.channel]
+          XCTAssertNotNil(channel)
+          let userPresence = channel?.uuids.first(where: { $0.uuid == configuration.uuid })
+          XCTAssertNotNil(userPresence?.state)
+        case let .failure(error):
+          XCTFail("Failed due to error: \(error)")
+        }
+        hereNowExpect.fulfill()
+      }
+    }
+
+    let listener = SubscriptionListener()
+    listener.didReceivePresence = { event in
+      if event.channel == self.channel, event.join.contains(configuration.uuid) {
+        performHereNow()
+      }
+    }
+    let token = client.add(listener)
+
+    client.subscribe(to: [channel], withPresence: true, setting: ["StateKey": "StateValue"])
+
+    defer { token.cancel() }
+    wait(for: [hereNowExpect], timeout: 10.0)
+  }
+
+  func testHereNow_SingleChannel_EmptyPresence() {
+    let hereNowExpect = expectation(description: "Here Now Response")
+
+    let configuration = PubNubConfiguration(from: testsBundle)
+    let client = PubNub(configuration: configuration)
+
+    client.hereNow(on: [emptyChannel], also: true) { result in
+      switch result {
+      case let .success(response):
+        XCTAssertTrue(response.channels.isEmpty)
+      case let .failure(error):
+        XCTFail("Failed due to error: \(error)")
+      }
+      hereNowExpect.fulfill()
+    }
+
+    wait(for: [hereNowExpect], timeout: 10.0)
+  }
+
+  // MARK: - Here Now Mutlti Channel
+
+  func testHereNow_MultieChannel_Stateless() {
```

Comment:
Typo: `MultieChannel` => `MultiChannel`

Reply:
Apparently I was unpacking my clothes around the time of writing these tests...